core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -684,17 +689,16 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
   }
 
   private def checkpointBrokerId(brokerId: Int) {
```

Comment:
Can you explain why we should let broker exit if broker id load fail? My concern is that this will cause broker to exit if any log directory goes offline, which defeats the purpose of this KIP.